--- ros2/ament.bzl.orig	2025-10-28 22:31:59.543714955 +0000
+++ ros2/ament.bzl	2025-10-28 23:02:50.641780766 +0000
@@ -179,6 +179,15 @@
             target_file = plugin.dynamic_library,
         )
         outputs.append(dynamic_library)
+        # Also create symlink for introspection typesupport (needed for services)
+        introspection_library = ctx.actions.declare_file(
+            paths.join(prefix_path, "lib", "lib" + package_name + "__rosidl_typesupport_introspection_cpp.so"),
+        )
+        ctx.actions.symlink(
+            output = introspection_library,
+            target_file = plugin.dynamic_library,
+        )
+        outputs.append(introspection_library)
 
     idls_from_deps = depset(
         transitive = [
@@ -200,17 +209,17 @@
             registered_packages.append(package_name)
         idl_manifest_contents = []
         for src in idl.srcs:
-            if src.extension != "msg":
+            if src.extension not in ["msg", "srv", "action"]:
                 continue
             src_file = ctx.actions.declare_file(
-                paths.join(prefix_path, "share", package_name, "msg", src.basename),
+                paths.join(prefix_path, "share", package_name, src.extension, src.basename),
             )
             ctx.actions.symlink(
                 output = src_file,
                 target_file = src,
             )
             outputs.append(src_file)
-            idl_manifest_contents.append(paths.join("msg", src.basename))
+            idl_manifest_contents.append(paths.join(src.extension, src.basename))
 
         idl_manifest = ctx.actions.declare_file(
             paths.join(prefix_path, _RESOURCE_INDEX_PATH, "rosidl_interfaces", package_name),
